src/bblocks/datacommons_tools/custom_data/data_management.py

```diff
@@ -869,26 +898,85 @@ def validate_config(self) -> CustomDataManager:
         self._config.validate_config()
         return self
 
-    def __repr__(self) -> str:
-        input_files_count = len(self._config.inputFiles)
-        sources_count = len(self._config.sources)
-        nodes_count = sum(
-            [len(var) for var in [n.nodes for n in self._mcf_nodes.values()] if var]
-        )
+    def merge_config(
+        self,
+        config: Config | dict | str | PathLike[str],
+        *,
+        policy: DuplicatePolicy = "error",
+    ) -> CustomDataManager:
+        """Merge ``config`` into the current configuration.
 
-        variables_count = len(self._config.variables or {}) + nodes_count
-        dataframes_count = len(self._data)
+        Args:
+            config: The config to merge. This can be a Config object, a dictionary,
+                or a path to a JSON file.
+            policy: How to resolve collisions. Can be "error", "override", or "ignore".
```

Comment:
There should be a bit more information on what the policies actually do so there is no confusion.

Reply:
Good idea, will add!